examples/log-mpi.py

```diff
@@ -10,6 +11,27 @@
 from mpi4py import MPI
 
 
+class UserLogQuantity(LogQuantity):
+    """Logging support for arbitrary user quantities."""
```

Comment:
I'd vote to lose the `user_function` here. The relationship of `set_quantity_value` and the `user_function` is not documented and not at all clear. If the `user_function` thing is needed, then that should probably a separate `LogQuantity`.